contracts/InterchainTokenFactory.sol

```diff
@@ -183,9 +257,27 @@ contract InterchainTokenFactory is IInterchainTokenFactory, ITokenManagerType, M
 
         if (minter != address(0)) {
             if (!token.isMinter(minter)) revert NotMinter(minter);
-            if (minter == address(interchainTokenService)) revert InvalidMinter(minter);
 
-            minter_ = minter.toBytes();
+            if (destinationMinter.length > 0) {
+                DeployApproval memory approval = DeployApproval({
+                    minter: minter,
+                    deployer: msg.sender,
+                    salt: salt,
+                    destinationChain: destinationChain,
+                    destinationMinter: destinationMinter
+                });
+                _useDeployApproval(approval);
+                minter_ = destinationMinter;
+            } else {
+                minter_ = minter.toBytes();
+            }
+
+            // Sanity check to prevent accidental use of the current ITS address as the destination minter
+            bytes memory itsAddress = address(interchainTokenService).toBytes();
+            if (keccak256(minter_) == keccak256(itsAddress)) revert InvalidMinter(minter);
```

Comment:
I think the one-liner is self-explanatory, but two lines are also ok with me.
```
// Sanity check to prevent accidental use of the current ITS address as the destination minter
if (keccak256(minter_) == keccak256(address(interchainTokenService).toBytes())) revert InvalidMinter(minter);
```